homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,260 @@
+}
+
+AIR_CLEANING_STATUS_MAP = {
+    0: "idle",
+    1: "idle",
+    2: "on",
+    3: "off",
+}
+
+FAN_STATUS_MAP = {0: "off", 1: "on"}
+
+
+@dataclass(frozen=True, kw_only=True)
+class AprilaireSensorDescription(SensorEntityDescription):
+    """Class describing Aprilaire sensor entities."""
+
+    status_key: str | None
+    status_sensor_available_value: int | None
+    status_sensor_exists_values: list[int] | None
+    value_key: str
+    value_fn: Callable[[Any, Any], StateType] | None
+
+
+SENSOR_TYPES: tuple[AprilaireSensorDescription, ...] = (
+    AprilaireSensorDescription(
+        key="indoor_humidity_controlling_sensor",
+        translation_key="indoor_humidity_controlling_sensor",
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        status_key=Attribute.INDOOR_HUMIDITY_CONTROLLING_SENSOR_STATUS,
+        status_sensor_available_value=0,
+        status_sensor_exists_values=[0, 1, 2],
+        value_key=Attribute.INDOOR_HUMIDITY_CONTROLLING_SENSOR_VALUE,
+        value_fn=None,
+    ),
+    AprilaireSensorDescription(
+        key="outdoor_humidity_controlling_sensor",
+        translation_key="outdoor_humidity_controlling_sensor",
+        device_class=SensorDeviceClass.HUMIDITY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        status_key=Attribute.OUTDOOR_HUMIDITY_CONTROLLING_SENSOR_STATUS,
+        status_sensor_available_value=0,
+        status_sensor_exists_values=[0, 1, 2],
+        value_key=Attribute.OUTDOOR_HUMIDITY_CONTROLLING_SENSOR_VALUE,
+        value_fn=None,
+    ),
+    AprilaireSensorDescription(
+        key="indoor_temperature_controlling_sensor",
+        translation_key="indoor_temperature_controlling_sensor",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=None,
+        status_key=Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_STATUS,
+        status_sensor_available_value=0,
+        status_sensor_exists_values=[0, 1, 2],
+        value_key=Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE,
+        value_fn=lambda sensor, value: convert_temperature_if_needed(
+            sensor.unit_of_measurement, value
+        ),
+    ),
+    AprilaireSensorDescription(
+        key="outdoor_temperature_controlling_sensor",
+        translation_key="outdoor_temperature_controlling_sensor",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        status_key=Attribute.OUTDOOR_TEMPERATURE_CONTROLLING_SENSOR_STATUS,
+        status_sensor_available_value=0,
+        status_sensor_exists_values=[0, 1, 2],
+        value_key=Attribute.OUTDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE,
+        value_fn=lambda sensor, value: convert_temperature_if_needed(
+            sensor.unit_of_measurement, value
+        ),
```

Comment:
may be sensitive to value = None, consider turnery 

```
value_fn=lambda sensor, value: None if value is None else convert_temperature_if_needed(
            sensor.unit_of_measurement, value
        )
```

Reply:
@EvanSchalton Mentioned above, but `convert_temperature_if_needed` does handle `None` so this should be fine.